zkevm-circuits/src/evm_circuit/execution/error_return_data_oo_bound.rs

```diff
@@ -84,13 +83,23 @@ impl<F: Field> ExecutionGadget<F> for ErrorReturnDataOutOfBoundGadget<F> {
             return_data_length.expr(),
             from_bytes::expr(&end.cells[..N_BYTES_U64]),
         );
-        // Any of [offset_out_of_range, end_out_of_range, end_exceed_length] occurs.
-        cb.require_in_set(
-            "Any of [offset_out_of_range, end_out_of_range, end_exceed_length] occurs",
-            not::expr(is_data_offset_within_range.expr())
-                + not::expr(is_end_within_range.expr())
-                + is_end_exceed_length.expr(),
-            vec![1.expr(), 2.expr(), 3.expr()],
+
+        // `end > U256::MAX` is necessary to check with AddWordsGadget carry. Since
+        // offset plus length may exceed a Word and remainder (end) may be less
+        // than `u64::MAX` (e.g. Word::MAX + 1).
+        cb.require_equal(
+            "Any of [offset > u64::MAX, end > U256::MAX, remainder_end > u64::MAX, remainder_end > return_data_length] occurs",
+            or::expr([
+                // offset > u64::MAX
+                not::expr(is_data_offset_within_range.expr()),
+                // end > U256::MAX
+                sum.carry().as_ref().unwrap().expr(),
+                // remainder_end > u64::MAX
```

Comment:
how about give `sum.carry().as_ref().unwrap().expr()` an intermediate variable `is_end_no_overflow`. In this case, maybe rename other `end` with `remainder_end` to match with naming in coding

Reply:
Added a new variable in commit [4ea268b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1321/commits/4ea268b414ea757e452eab267e9a8a656c0cadcc#diff-2f2120439adc744a42a4ca326eedbd453e6b8321cce2105a2e64741e17b67cedR74) (if `carry == 1`, `data_offset + size` is U256 overflow, and it is also an error of return data out of bound):
```
let is_end_u256_overflow = sum.carry().as_ref().unwrap();
```

And rename other variables with `end` to `remainder_end`.